### PR TITLE
Support identifier clause for Databricks

### PR DESCRIPTION
**Pull request by PaulBurridge**
Provides support for IDENTIFIER clause.

### Brief summary of the change made

Supports IDENTIFIER(string or string expression) in stead of database, schema, table or view names as per [documentation].(https://docs.databricks.com/en/sql/language-manual/sql-ref-names-identifier-clause.html)

### Are there any other side effects of this change that we should be aware of?

Not that I am aware of apologies if I have missed something as this is my first PR.

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_databricks.py [hunk 1]
```diff
@@ -348,7 +348,63 @@
...
+class TableReferenceSegment(ObjectReferenceSegment):
+    """A reference to an table, CTE, subquery or alias."""
+
+    type = "table_reference"
+
+
+class SchemaReferenceSegment(ObjectReferenceSegment):
+    """A reference to a schema."""
+
+    type = "schema_reference"
```
Comment (alanmcruickshank) [inline src/sqlfluff/dialects/dialect_databricks.py:391]: I don't think these need redefining? I can't see any edits vs the classess created in the ansi dialect, so no need to redefine. Unless I've missed something?

**Review by alanmcruickshank — COMMENTED** [review]
I think you've redefined a couple of objects unnecessarily - otherwise this looks good 👍 .

Thanks for the contribution

File: src/sqlfluff/dialects/dialect_databricks.py (hunk 1 above)
Comment (PaulBurridge) [reply src/sqlfluff/dialects/dialect_databricks.py:391]: @alanmcruickshank 

These were needed to inherit changes to `ObjectReferenceSegment` which has been updated, without the tests fail.

Comment (alanmcruickshank) [reply src/sqlfluff/dialects/dialect_databricks.py:391]: iiiiiineresting. Good catch. Not super helpful from a development point of view, but I guess that's what it is right now.

Thanks for the contribution!

**Pull request merged**